Shopify/cli
# Introduce support to password protected stores

### WHY are these changes introduced?

Currently, the session object in the proxy does not include the password flag.

### WHAT is this pull request doing?

This PR reuses the `storefront-password-prompt` module (from `shopify theme console`) to prompt users for the password flag.

### How to test your changes?

- Open this URL to activate the password in your store: `https://admin.shopify.com/store/<your_store>/online_store/preferences?tutorial=unlock`
- Set a password
- Run `shopify theme dev --dev-preview`
- Notice the proxy works with your store

### Post-release steps

N/A

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
